Bluetooth: Convert to new logger subsystem

jhedberg opened the pull request:
Convert the Bluetooth subsystem to the new logger APIs.

Known issues:

 - Some bt_hex() uses (especially the longer chunks of data) should be replaced with `LOG_HEXDUMP()`
 - BT_* should be renamed to LOG_* but this depends on function name support (see #10644)

Commit:
Bluetooth: Switch from SYS_LOG to logger-based logging

Initial conversion to use syslog instead of logger.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
Bluetooth: Only build bt_hex() if BT_DEBUG is enabled

bt_hex() should be included in the same block as the address helpers.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
Bluetooth: log: Make address helpers more robust

Use irq_lock (same way as bt_hex does it) and increase the number of
static buffers to reduce the risk of reuse before a buffer gets
processed.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
Bluetooth: Remove custom stack macros

Now that log processing happens in a separate thread, the
BT_STACK_EXTRA macro is not needed (since there's no significant
overhead), and therefore the BT_STACK macros become unnecessary as
well.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

dleach02 commented:
There appears to be a lot more changes here outside of the logging. Is the intent to separate the other changes and have a logging only change made available at some point?

Commit:
Bluetooth: Convert string helpers to use log_strdup

This reduces the multiple buffer requirement to a single buffer.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

jhedberg commented on subsys/bluetooth/host/monitor.c:
```diff
@@ -241,6 +216,108 @@ extern void __stdout_hook_install(int (*fn)(int));
+		 * not to add these to the end of messages.
+		 */
+		if (data[i] == '\r' || data[i] == '\n') {
+			break;
```
@nordic-krch could we get a flag for the log subsystem so that I don't need to do this kind of truncation?

Commit:
Bluetooth: Convert monitor protocol to a logger backend

Convert the monitor protocol to a proper logger backend. This also
means that our log.h headerfile gets greatly simplified.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
samples: boards: mesh/onoff-app: Remove SYS_LOG usage

SYS_LOG is being deprecated. Use printk instead.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

jhedberg commented:
I've removed the DNM tag since this now builds successfully and the monitor back end works. Strings are also handled using log_strdup() however some bt_hex calls should probably still be converted to LOG_HEXDUMP(). I'd also like to rename all BT_* calls to corresponding LOG_* ones, however that may require waiting for a resolution for #10644 since e.g. BT_DBG and BT_ERR include the function name in the output.

nashif commented:
looks like you missed a few samples/tests?

```
samples/bluetooth/hci_spi/prj.conf
samples/bluetooth/hci_spi/src/main.c
samples/bluetooth/hci_uart/src/main.c
samples/bluetooth/ipsp/prj.conf
samples/bluetooth/ipsp/prj_dbg.conf
samples/bluetooth/ipsp/prj_zep1656.conf
samples/bluetooth/ipsp/src/main.c
samples/boards/nrf52/mesh/onoff-app/prj.conf
tests/bluetooth/tester/src/bttester.c
tests/bluetooth/tester/src/bttester.h
tests/bluetooth/tester/src/gap.c
tests/bluetooth/tester/src/gatt.c
tests/bluetooth/tester/src/l2cap.c
tests/bluetooth/tester/src/mesh.c
```

jhedberg commented:
> looks like you missed a few samples/tests?

Yep, I was mainly focusing on the stack rather than sample apps and missed these. I'll work on getting them converted.

Commit:
samples: hci_spi: Convert from SYS_LOG to new logger

Use the LOG_* macros instead of the deprecated SYS_LOG_* ones.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
samples: hci_uart: Convert from SYS_LOG to new logger

Use the LOG_* macros instead of the deprecated SYS_LOG_* ones.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
samples: ipsp: Convert from SYS_LOG to new logger

Use the LOG_* macros instead of the deprecated SYS_LOG_* ones.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
tests: bluetooth/tester: Convert from SYS_LOG to new logger

Use the LOG_* macros instead of the deprecated SYS_LOG_* ones.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

jhedberg commented:
@nashif I've updated the apps now

carlescufi reviewed (changes requested):
We are still using `printk` in the built-in samples/bluetooth/*. Is this intended? it looks pretty ugly to me:

```
***** Booting Zephyr OS zephyr-v1.13.0-1191-gfbe43203b2 *****
Bluetooth initialized
Advertising successfully started
[00:00:00.007,629] <inf> bt_hci_core: HW Platform: Nordic Semiconductor (0x0002)
[00:00:00.007,659] <inf> bt_hci_core: HW Variant: nRF52x (0x0002)
[00:00:00.007,659] <inf> bt_hci_core: Firmware: Standard Bluetooth controller (0x00) Version 1.13 Build 99
[00:00:00.007,904] <wrn> bt_hci_core: bt_init: No ID address. Expecting one to come from storage.
[00:00:00.013,336] <inf> bt_hci_core: Identity: fb:d0:38:54:d1:7a (random)
[00:00:00.013,336] <inf> bt_hci_core: HCI: version 5.0 (0x09) revision 0x0000, manufacturer 0x05f1
[00:00:00.013,336] <inf> bt_hci_core: LMP: version 5.0 (0x09) subver 0xffff
```

carlescufi reviewed (approved):
Seems like using `printk` for sample output is the standard behavior right now in Zephyr, so no reason for me to nack this PR for this.
